dbeam/src/main/scala/com/spotify/dbeam/PsqlAvroJob.scala

```diff
@@ -65,21 +65,26 @@ object PsqlAvroJob {
     }
   }
 
-  def validateReplication(partition: DateTime, lastReplication: DateTime): DateTime = {
-    if (lastReplication.isBefore(partition)) {
-      log.error(s"Replication was not completed for partition, " +
-        s"expected >= $partition, actual = $lastReplication")
-      System.exit(20)
-    }
-    partition
+  def isReplicationDelayed(partition: DateTime,
+                           lastReplication: DateTime,
+                           partitionPeriod : ReadablePeriod): Boolean = {
+    lastReplication.isBefore(partition.plus(partitionPeriod))
   }
 
   @SuppressWarnings(Array("org.wartremover.warts.OptionPartial"))
   def main(cmdlineArgs: Array[String]): Unit = {
     val (sc: ScioContext, args: JdbcExportArgs) = JdbcExportArgs.contextAndArgs(cmdlineArgs)
     validateOptions(args)
 
-    validateReplication(args.partition.get, queryReplication(args.createConnection()))
+    val partition = args.partition.get
+    val lastReplication = queryReplication(args.createConnection())
+    val partitionPeriod: ReadablePeriod = args.partitionPeriod
+
+    if (isReplicationDelayed(partition, lastReplication, partitionPeriod)) {
+      log.error(s"Replication was not completed for partition, " +
+        s"expected >= $partition, actual = $lastReplication")
```

Comment:
it should be `partition.plus(partitionPeriod)`

Reply:
And after having a second look I wonder do we really need the `isReplicationDelayed` method? It  's supposed to hide the validation logic but we exposing it in the log message anyway. 
The alternative would be to put the log msg back into method. What do you think?

Reply:
I agree. I think it’s safe to move the error log to the `isReplicationDelayed` method.